pkg/algorithm/cstorpoolselect/v1alpha1/select.go

```diff
@@ -103,6 +113,93 @@ type policy interface {
 	filter(*csp.CSPList) (*csp.CSPList, error)
 }
 
+// scheduleWithOverProvisioningAwareness is a pool
+// selection implementation.
+type scheduleWithOverProvisioningAwareness struct {
+	overProvisioning bool
+	spcName          string
+	volumeNamespace  string
+	capacity         resource.Quantity
+	err              []error
+}
+
+// priority returns the priority of the
+// policy implementation
+func (p scheduleWithOverProvisioningAwareness) priority() priority {
+	return highPriority
+}
+
+// name returns the name of the policy
+// implementation
+func (p scheduleWithOverProvisioningAwareness) name() policyName {
+	return overProvisioningPolicy
+}
+
+// filter selects the pools available on the host
+// for which the policy has been applied
+func (p scheduleWithOverProvisioningAwareness) filter(pools *csp.CSPList) (*csp.CSPList, error) {
+	if len(p.err) > 0 {
+		return nil, errors.Errorf("failed to fetch overprovisioning details:%v", p.err)
+	}
+
+	if p.overProvisioning {
+		klog.Info("Overprovisioning restriction policy not added as overprovisioning is enabled")
+		return pools, nil
+	}
+
+	filteredPools := csp.ListBuilder().List()
+	for _, pool := range pools.Items {
+		volCap, err := getAllVolumeCapacity(pool.Object)
```

Comment:
can we rename this variable to make this function more readable. e.g. `consumedCapacity`